### ISSUE TITLE
Request for additional features to GET /ob/chatconversations

### ISSUE THREAD

**Issue by drwasho** [OpenBazaar/openbazaar-go#1416]
Currently there is no easy way to fetch a list of chat conversations related to an order, and group chat conversations (typically related to an order).

In Haven for example, we have a dedicated view set aside to display order related chats so that users can jump into any new developments (without having to dig into the order views). 

The only way to get a list of new chat messages (or conversations) related to an order is to make calls to fetch purchases and sales list, parse the response to look for unread messages/conversations for each order... bottomline there's a lot of unnecessary overhead that needs to be written on the client-side.

When we make a `GET /ob/chatconversations` call, by default it returns all conversations where the `subject` line is empty. This is fine, and there's no reason to chat that. **What I propose** is that we include a filter called `subject=true` that returns all chat conversations that have a subject. If `subject=false` or if `subject` is left out of the URL, revert to the default behaviour of returning chat conversations without a subject.

The response for `GET /ob/chatconversations` should therefore include the `subject` field (which may or may not be an empty string):

```JSON
[
    {
        "lastMessage": "Ola1",
        "outgoing": true,
        "peerId": "QmPyweNFHayJgBRqYaoBatmnfeonxtuVZvQjtWymzmwkav",
        "subject": "QmWYHtr9xifn46BLyc8iz6ivViJ5ghJEGNvcaBGzoWFjNs",
        "timestamp": "2018-12-19T07:36:00Z",
        "unread": 0
    }
]
```

Furthermore, for those chat conversations that are _group chats_, can we modify the response to show that it is a group chat in the following way :

```JSON
[
    {
        "lastMessage": "Ola1",
        "outgoing": true,
        "peerId": "QmPyweNFHayJgBRqYaoBatmnfeonxtuVZvQjtWymzmwkav",
        "subject": "QmWYHtr9xifn46BLyc8iz6ivViJ5ghJEGNvcaBGzoWFjNs",
        "timestamp": "2018-12-19T07:36:00Z",
        "unread": 0
    },
    {
        "lastMessage": "Yo",
        "outgoing": true,
        "peerId": ["QmQuaraGBJM8GheDtzwscxhK9HaA2t4JqRFjLePijgXxV3", "QmQuaraGBJM8GheDtzwscxhK9HaA2t4JqRFjLePijgXxV3", "QmactsH18iDPCyBEpsrP4RmCg2MJpKW98SBfWNctwCqzJW"],
        "subject": "QmYTWWnCdjTENUmK9d9ToUysnKmHDz9WFmiQZ9b6PmHGss",
        "timestamp": "2018-12-19T07:27:44Z",
        "unread": 0
    }
]
```

^^ so if it is a group chat, the `peerId` field is an array of all the participants

**Comment by placer14** [comment]
It sounds like you want to get a list of unread messages regardless of what subject they are associated with? (one peer, a group of peers or an order) That's doable, just need to review the state of the chat API.

One general thought: I'm not a fan of having different types being returned for the same field (peerID being an array or a string). This makes consuming logic around the API cumbersome. I would probably make it always an array to show the list of participants which could also be just one peer.

**Comment by drwasho** [comment]
> It sounds like you want to get a list of unread messages regardless of what subject they are associated with?

Not exactly, I want to get a full list of conversations (i.e. with and without subjects).

> I would probably make it always an array to show the list of participants which could also be just one peer.

That works!
